plugins/filter/filters.py

```diff
@@ -297,39 +316,41 @@ def client_properties(self, listener_dict, default_ssl_enabled, bouncy_castle_ke
             final_dict[config_prefix + 'ssl.keystore.type'] = 'BCFKS'
             final_dict[config_prefix + 'ssl.truststore.type'] = 'BCFKS'
 
-        if self.normalize_sasl_protocol(listener_dict.get('sasl_protocol', default_sasl_protocol)) == 'PLAIN' and not omit_jaas_configs:
+        sasl_protocols = listener_dict.get('sasl_protocol', default_sasl_protocol)
+        normalize_sasl_protocols = self.normalize_sasl_protocol(sasl_protocols)
+        if 'PLAIN' in normalize_sasl_protocols and not omit_jaas_configs:
```

Comment:
how are we maintaining the preference for client protocol in this case? As per my understanding, we need to create clients based on protocol[0] but here we will just overwrite based on this function

Reply:
Agreed. We should not overwrite the configs and consider the first value when adding client config.